sdk/core/core/inc/az_callback.h

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef AZ_CALLBACK_H
+#define AZ_CALLBACK_H
+
+#include <_az_cfg_prefix.h>
+
+typedef void * az_callback_data;
+
+#define AZ_CAT(A, B) A##B
```

Comment:
given that we are shipping source code, please add a comment block to each file with a short blurb description, similar to other Azure SDK languages